text/0026-semantic-versioning.md

```diff
@@ -0,0 +1,201 @@
+| Removal of an order group dropping<br>support of some workflow | Addition of an order group that do<br>not cause change in detection logic |  |
+| Swapping order of order groups | Addition of a component buildpack |  |
+| Addition of an order group<br>causing change in detection logic |  |  |
+| Removal of a component buildpack |  |  |
+<br/>
+
+
+#### Builder
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| Lifecycle major bumps | Minor bumps of components | Patch bumps of components |
+| Stack Image major bumps | Addition of a buildpack<br>as last in the order |  |
+| Buildpack major bumps |  |  |
+| Change in Stack ID |  |  |
+| Removal of a Buildpack |  |  |
+| Swapping order of buildpacks |  |  |
+| Addition of a buildpack except<br>as last in the order |  |  |
+<br/>
+
+#### Stack
+| `Major`                 | `Minor`                   | `Patch`                   |
+|-------------------------|---------------------------|---------------------------|
+| Removal of  a package   | Addition of a new package | Update of a package       |
+
+### Intial Development
+
+A version of `0.0.x` is considered to be in initial development and may not be
+suitable for user consumption. When in initial development, semantic versioning
+guidelines do not apply.
+
+## User Policy
+
+* **Patch** bump: User should update their buildpack/builder without
+	hesitation.
+
+* **Minor** bump: User can update their buildpack/builder to avail new
+	features. Nothing will break if buildpack-set default dependency versions are
+	not overridden.
+
+* **Major** bump: Users must review changelog and test build system on update.
+	May contain breaking changes.
+
+
+### Critical security updates
+
+If a user is on the latest `Major` and `Minor` versions, critical security
+updates when they occur should be available for users on the latest `Patch`
+version.
+
+## Process
+
+This section proposes a process that will help maintainers easily determine
+whether a given change or a set of changes require a bump in the major, minor,
+or patch component.
+
+* Changes should be accepted to the main line of code only via Pull Requests.
+
+* All Pull Requests (including automated PRs by the
+  [paketo-bot](https://github.com/paketo-bot)) must be tagged with either of
+  the 3 labels:
+	* `semver:major` - denoting accepting the change will require a major
+	  version bump in the next release.
+	* `semver:minor` - denoting accepting the change will require a minor
+	  version bump in the next release.
+	* `semver:patch` - denoting accepting the change will require a patch
+	  version bump in the next release.
+
+* When releasing the next version, the maintainer must pick the most
+  significant required version bump from among the accepted Pull Requests and
+  bump that component of the release. The order of significance is `major` >
+  `minor` > `patch`.
+
+* Pull requests with critical security updates must be additionally tagged as
```

Comment:
Under our current branch management and PR process, tagging something `security-fix` and immediately releasing as a patch could roll in changes on `main` that warrant a minor or major bump.

Is this RFC advocating for a new branch management process where PRs are merged into major/minor/patch branches?

Reply:
No, this part is simply standardizing a label for critical security fixes. See User Policy section - If a user is on the latest Major and Minor versions, critical security updates when they occur should be available for users on the latest Patch version.
Assume there is a PR that removes PHP 7.5 and adds PHP 7.6. This would be naturally labeled as `semver:minor`. But then if it is known that PHP 7.5 has a security vulnerability that is fixed by 7.6, then the maintainer must release this as a patch due to the user policy commitment stated above. In such a case, Pull Request creator or whoever responsible can tag it as `security-fix` so that a patch can be released.